sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContext.cs

```diff
@@ -191,14 +277,32 @@ public partial interface IDynamoDBContext : IDisposable
 
         #region Transact Write
 
+        /// <summary>
+        /// Creates a strongly-typed TransactWrite object, allowing
+        /// a transactional write operation against DynamoDB.
+        /// </summary>
+        /// <typeparam name="T">Type of objects to write.</typeparam>
+        /// <returns>Empty strongly-typed TransactWrite object.</returns>
+        TransactWrite<T> CreateTransactWrite<T>();
+
         /// <summary>
         /// Creates a strongly-typed TransactWrite object, allowing
         /// a transactional write operation against DynamoDB.
         /// </summary>
         /// <typeparam name="T">Type of objects to write.</typeparam>
         /// <param name="operationConfig">Config object which can be used to override that table used.</param>
         /// <returns>Empty strongly-typed TransactWrite object.</returns>
-        TransactWrite<T> CreateTransactWrite<T>(DynamoDBOperationConfig operationConfig = null);
+        [Obsolete("Use the CreateTransactWrite overload that takes TransactWriteConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to CreateTransactWrite.")]
+        TransactWrite<T> CreateTransactWrite<T>(DynamoDBOperationConfig operationConfig);
```

Comment:
why did you change the signature of the method?

Reply:
See https://github.com/aws/aws-sdk-net/pull/3376#discussion_r1672274431